### ISSUE TITLE
error thrown by draw_key_pointrange() if any params value is NULL

### ISSUE THREAD

**Issue by corybrunson** [tidyverse/ggplot2#4967]
[The `draw_key_*()` function for `GeomPointrange`](https://github.com/tidyverse/ggplot2/blob/a58b48c961cb391b8646bf072b6620a0c9f3d999/R/legend-draw.r#L201) (as defined in the development version approaching release) uses `transform()` to add parameter values to the `data` being passed from the layer to the legend. When at least one parameter is set to `NULL`, this has the effect of throwing an error. I can't install the development version of **ggplot2** on the machine i'm using, but here is the error i get via `load_all()` on a local clone:

```r
devtools::load_all()
#> ℹ Loading ggplot2

df <- data.frame(
  trt = factor(c(1, 1, 2, 2)),
  resp = c(1, 5, 3, 4),
  group = factor(c(1, 2, 1, 2)),
  upper = c(1.1, 5.3, 3.3, 4.2),
  lower = c(0.8, 4.6, 2.4, 3.6)
)

p <- ggplot(df, aes(trt, resp, colour = group))
p + geom_pointrange(aes(ymin = lower, ymax = upper))

p +
  geom_line(aes(group = group)) +
  geom_pointrange(aes(ymin = lower, ymax = upper), fatten = 10)

p +
  geom_line(aes(group = group)) +
  geom_pointrange(aes(ymin = lower, ymax = upper), fatten = NULL)
#> Error in (function (..., row.names = NULL, check.rows = FALSE, check.names = TRUE, : 
#>  arguments imply differing number of rows: 0, 1
```

1. Is this effect intended? If so, should it be applied to the other `draw_key_*()` functions?
2. If unintended, it could be resolved as follows (or in some more elegant way, i'm sure):

```r
draw_key_pointrange <- function(data, params, size) {
  params[vapply(params, is.null, NA)] <- NULL
  grobTree(
    draw_key_linerange(data, params, size),
    draw_key_point(transform(data, params, size = (data$size %||% 1.5) * 4))
  )
}
```

I'm currently doing this in an extension package, see [here](https://github.com/corybrunson/ggalluvial/commit/43416893f691f7ee213b05e4e933ccfa3a8b3054).

Addendum: Here is an example of the root problem:

```r
transform(data.frame(a = 1, b = 2), list(c = NULL), d = 4)
#> Error in data.frame(list(a = 1, b = 2), list(c = NULL), d = 4) : 
#>   arguments imply differing number of rows: 1, 0
```

**Comment by teunbrand** [comment]
I think this issue has been fixed in the current release (3.4.0)

**Comment by corybrunson** [comment]
Confirmed! Thanks for checking back in.
